index.js

```diff
@@ -3,15 +3,25 @@
 var Stream = require('stream');
 var Path = require('path');
 
-function gulpRename(obj) {
+function gulpRename(obj, options) {
+
+  options = options || {};
 
   var stream = new Stream.Transform({objectMode: true});
 
   function parsePath(path) {
     var extname = Path.extname(path);
+    var dirname = Path.dirname(path);
+    var basename = Path.basename(path, extname);
+    if (options.multiExt) {
+      while (Path.extname(basename) !== '') {
```

Comment:
Could you avoid using a while loop here, and check if there any node built-ins that handle this? I would be curious to see how other people implement this.

Reply:
No node built-ins unfortunately. `path.extname` doesn't handle multiple extensions, that's why I used it in a loop at first.